addons/hr/models/hr_version.py

```diff
@@ -232,11 +232,16 @@ def _check_dates(self):
                     continue
                 if date_start <= contract_date_end and version.contract_date_start <= date_to:
                     raise ValidationError(_(
-                        'You have some overlapping contracts for %(employee)s:\n%(overlaps)s',
+                        'Overlapping contracts for %(employee)s:\n%(overlaps)s',
                         employee=version.employee_id.display_name,
                         overlaps='\n'.join(
-                            [f'Version ({version.display_name}): {version.contract_date_start} - {version.contract_date_end}'] +
-                            [f'Version ({version.display_name}): {date_start} - {date_end}' for version in versions])))
+                            [f'Version {format_date(version.env, version.date_version, date_format="MMM d, y")}: '
+                             f'from {format_date(version.env, version.contract_date_start, date_format="MMM d, y")} '
+                             f'to {format_date(version.env, version.contract_date_end, date_format="MMM d, y") if version.contract_date_end else "Indefinite"}'] +
+                            [f'Version {format_date(version.env, version.date_version, date_format="MMM d, y")}: '
+                             f'from {format_date(version.env, version.contract_date_start, date_format="MMM d, y")} '
+                             f'to {format_date(version.env, version.contract_date_end, date_format="MMM d, y") if version.contract_date_end else "Indefinite"}'
+                             for version in versions])))
```

Comment:
I think we could reduce the complexity that way :thinking:
Also we could have a var `all_version = (versions|version)` and loop through it

```suggestion
                            [f'Version {format_date(overlap_version.env, overlap_version.date_version, date_format="MMM d, y")}: '
                             f'from {format_date(overlap_version.env, overlap_version.contract_date_start, date_format="MMM d, y")} '
                             f'to {format_date(overlap_version.env, overlap_version.contract_date_end, date_format="MMM d, y") if overlap_version.contract_date_end else "Indefinite"}'
                             for overlap_version in (versions|version)])))
```